lib/src/query/filter.dart

```diff
@@ -1,35 +1,87 @@
 import 'dart:collection';
 
-class Filter with MapMixin<String, String> {
+class Filter with MapMixin<String, Object> {
   /// Example:
   /// ```dart
-  /// Filter({'post': '1,2', 'author': '12'}).addTo(url);
+  /// Filter({'post': '1,2', 'author': {'id': '12', 'role': 'admin'}}).addTo(url);
   /// ```
   /// encodes into
   /// ```
-  /// ?filter[post]=1,2&filter[author]=12
+  /// ?filter[post]=1,2&filter[author][id]=12&filter[author][role]=admin
   /// ```
-  Filter([Map<String, String> parameters = const {}]) {
+  Filter([Map<String, Object> parameters = const {}]) {
     addAll(parameters);
   }
 
-  static Filter fromUri(Uri uri) => Filter(uri.queryParametersAll
-      .map((k, v) => MapEntry(_regex.firstMatch(k)?.group(1) ?? '', v.last))
-    ..removeWhere((k, v) => k.isEmpty));
+  static Filter fromUri(Uri uri) {
+    Map<String, Object> filters = {};
```

Comment:
Might make it final and use the short initialization syntax.

Reply:
I can't see how we could use the short initialization syntax, as the function call mutates the filters map.